elyra/kfp/operator.py

```diff
@@ -0,0 +1,358 @@
+#
+# Copyright 2018-2021 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import string
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from kfp.dsl import ContainerOp
+from kubernetes.client.models import V1EmptyDirVolumeSource
+from kubernetes.client.models import V1EnvVar
+from kubernetes.client.models import V1EnvVarSource
+from kubernetes.client.models import V1ObjectFieldSelector
+from kubernetes.client.models import V1Volume
+from kubernetes.client.models import V1VolumeMount
+
+from elyra._version import __version__
+
+"""
+The ExtendedContainerOp uses a python script to bootstrap the user supplied image with the required dependencies.
+In order for the script run properly, the image used, must at a minimum, have the 'curl' utility available
+and have python3
+"""
+
+# Inputs and Outputs separator character.  If updated,
+# same-named variable in bootstrapper.py must be updated!
+INOUT_SEPARATOR = ';'
+
+ELYRA_GITHUB_ORG = os.getenv("ELYRA_GITHUB_ORG", "elyra-ai")
+ELYRA_GITHUB_BRANCH = os.getenv("ELYRA_GITHUB_BRANCH", "master" if 'dev' in __version__ else "v" + __version__)
+ELYRA_PIP_CONFIG_URL = os.getenv('ELYRA_PIP_CONFIG_URL', 'https://raw.githubusercontent.com/{org}/elyra/'
+                                                         '{branch}/etc/kfp/pip.conf'.
+                                                         format(org=ELYRA_GITHUB_ORG, branch=ELYRA_GITHUB_BRANCH))
+ELYRA_BOOTSTRAP_SCRIPT_URL = os.getenv('ELYRA_BOOTSTRAP_SCRIPT_URL', 'https://raw.githubusercontent.com/{org}/'
+                                                                     'elyra/{branch}/elyra/kfp/bootstrapper.py'.
+                                                                     format(org=ELYRA_GITHUB_ORG,
+                                                                            branch=ELYRA_GITHUB_BRANCH))
+ELYRA_REQUIREMENTS_URL = os.getenv('ELYRA_REQUIREMENTS_URL', 'https://raw.githubusercontent.com/{org}/'
+                                                             'elyra/{branch}/etc/kfp/requirements-elyra.txt'.
+                                                             format(org=ELYRA_GITHUB_ORG,
+                                                                    branch=ELYRA_GITHUB_BRANCH))
+
+
+class ExtendedContainerOp(ContainerOp):
```

Comment:
`ExtendedContainerOp` seems too obvious and not very informative.  How about `ElyraGenericOp` or `ElyraFileOp`.  Since, I suspect, nearly every operator extends `ContainerOp`, I don't think we need to preserve that aspect of things. `ElyraGenericOp` implies it's one of our generic operators, although, per the previous discussion, those may not always be file-based, as they are today - thus, the `ElyraFileOp`.

Reply:
+1, a little too on the nose

Reply:
I am not sure I wanted to use Elyra on the name of the op, would ExtendedFileOp or just FileOp be ok?

Reply:
I understand not wanting `Elyra` in the op name.  `FileOp` seems fine.  @ptitzler had suggested `RunFileOp` in a sidebar and I like that too and that's what happens.  I guess in that sense `ExecuteFileOp` also works.  I don't think `Extended` fits this context.